mlflow/models/evaluation/default_evaluator.py

```diff
@@ -466,14 +468,16 @@ def _evaluate_custom_metric(custom_metric_tuple, eval_df, builtin_metrics):
                                 ``custom_metrics`` parameter of ``mlflow.evaluate``
     :param eval_df: A Pandas dataframe object containing a prediction and a target column.
     :param builtin_metrics: A dictionary of metrics produced by the default evaluator.
+    :param additional_df: A Pandas dataframe that contains additional information for the overall dataset
+    :param additional_array: A numpy array that contains additional information for the overall dataset
     :return: A scalar metric value.
     """
     exception_header = (
         f"Custom metric '{custom_metric_tuple.name}' at index {custom_metric_tuple.index}"
         " in the `custom_metrics` parameter"
     )
 
-    metric = custom_metric_tuple.function(eval_df, builtin_metrics)
+    metric = custom_metric_tuple.function(eval_df, builtin_metrics, additional_df, additional_array)
```

Comment:
I think this is a breaking change. @jinzhang21 What do you think ?

Reply:
it actually is, after running `pytest` across mlflow. when i tested it locally, i did it for `models/evaluation/default_evaluator.py`. 

anywhere with `custom_metric` will need to be changed to 4 inputs.

Reply:
We should avoid breaking change, otherwise some 3rd-party lib written custom metrics might not work on new mlflow version.

Reply:
alright sure, this makes sense.